pkg/prototype/values.go

```diff
@@ -0,0 +1,224 @@
+// Copyright 2018 The ksonnet authors
+//
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package prototype
+
+import (
+	"bufio"
+	"io"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+	"github.com/spf13/pflag"
+)
+
+const (
+	// OptionValuesFile is flag key containing the path to a values file
+	OptionValuesFile = "values-file"
+)
+
+// Values extracts values for a prototype from supplied flags.
+func Values(fs afero.Fs, p *Prototype, flags *pflag.FlagSet) (map[string]string, error) {
+	if p == nil {
+		return nil, errors.New("prototype is required")
+	}
+
+	if flags == nil {
+		return nil, errors.New("flags is required")
+	}
+
+	valuesFile, err := flags.GetString(OptionValuesFile)
+	if err != nil {
+		return nil, err
+	}
+
+	name, err := flags.GetString("name")
+	if err != nil {
+		return nil, errors.New("name for prototype is required")
+	}
+
+	if valuesFile != "" {
+		vff, err := newValuesFromFile(fs, p, name, valuesFile)
```

Comment:
We should document somewhere that valuesFile will take precedence over bare flags.